pkg/operator/controller/controller_test.go

```diff
@@ -1459,3 +1474,148 @@ func createReconciler(client realClient.Client) *ReconcileCDI {
 
 	return r
 }
+
+func validateEvents(events chan string, match map[string]bool) {
+	// Closing the channel allows me to do non blocking reads of the channel, once the channel runs out of items the loop exits.
+	close(events)
+	for event := range events {
+		val, ok := match[event]
+		Expect(ok).To(BeTrue(), "Event [%s] was not expected", event)
+		if !val {
+			match[event] = true
+		}
+	}
+	for k, v := range match {
+		Expect(v).To(BeTrue(), "Event [%s] not observed", k)
+	}
+}
+
+func createDeleteCDIAfterReadyEventValidationMap() map[string]bool {
+	match := createReadyEventValidationMap()
+	match["Normal DeleteResourceStart Started deleting deployment cdi-deployment"] = false
+	match["Normal DeleteResourceSuccess Deleted deployment cdi-deployment successfully"] = false
+	match["Normal DeleteResourceStart Started deleting worker resources"] = false
+	match["Normal DeleteResourceSuccess Deleted worker resources successfully"] = false
+	return match
+}
+
+func createDeleteCDIEventValidationMap() map[string]bool {
+	match := createNotReadyEventValidationMap()
+	match["Normal DeleteResourceStart Started deleting deployment cdi-deployment"] = false
+	match["Normal DeleteResourceSuccess Deleted deployment cdi-deployment successfully"] = false
+	match["Normal DeleteResourceStart Started deleting worker resources"] = false
+	match["Normal DeleteResourceSuccess Deleted worker resources successfully"] = false
+	return match
+}
+
+func createErrorCdiEventValidationMap() map[string]bool {
+	match := createNotReadyEventValidationMap()
+	match["Warning ConfigError Reconciling to error state, unwanted CDI object"] = false
+	return match
+}
+
+func createReadyEventValidationMap() map[string]bool {
+	match := createNotReadyEventValidationMap()
+	match["Normal CreateResourceStart Ensuring upload proxy route exists"] = false
+	match["Normal CreateResourceSuccess Successfully ensured upload proxy route exists"] = false
+	match["Normal DeployCompleted Deployment Completed"] = false
+	return match
+}
+
+func createNotReadyEventValidationMap() map[string]bool {
+	// match is map of strings and if we observed the event.
+	// We are not interested in the order of the events, just that the events happen at least once.
+	match := make(map[string]bool)
+	match["Normal DeployStarted Started Deployment"] = false
+	match["Normal CreateResourceStart Started creation of resource *v1.ClusterRole cdi-apiserver"] = false
+	match["Normal CreateResourceSuccess Successfully created resource *v1.ClusterRole cdi-apiserver"] = false
```

Comment:
can we use const string prefixes for easier read/maint? e.g.
const normalCreateResourceStart string = "Normal CreateResourceStart Started creation of resource"